codalab/apps/newsletter/views.py

```diff
@@ -50,19 +54,25 @@ def newsletter_signup(request):
 
             _send_mail(data, html_file=html_template, text_file=email_message, subject=subject, to_email=to_email)
 
-    else:
-        messages.warning(request, 'Please use a valid email address to subscribe to the newsletter',
-                         'alert alert-warning alert-dismissible')
-
     context = {
         'form': form,
     }
 
     template = "newsletter/signup.html"
+
+    storage = messages.get_messages(request)
```

Comment:
whoa what is this doing?!

Reply:
Removes stale message from the cache. I removed the unnecessary loop in the latest push.